packages/openid4vp/src/openid4vp-auth-request/parse-openid4vp-auth-request-params.ts

```diff
@@ -0,0 +1,54 @@
+import { Oauth2Error, decodeJwt } from '@openid4vc/oauth2'
+import { uriDecodeObject } from '@openid4vc/utils'
+import * as v from 'valibot'
+import { type JarAuthRequest, vJarAuthRequest } from '../jar/v-jar-auth-request'
+import { type Openid4vpAuthRequest, vOpenid4vpAuthRequest } from './v-openid4vp-auth-request'
+
+export interface ParsedJarOpenid4vpAuthRequest {
+  type: 'jar'
+  provided: 'uri' | 'jwt' | 'params'
+  params: JarAuthRequest
+}
+
+export interface ParsedOpenid4vpAuthRequest {
+  type: 'openid4vp'
+  provided: 'uri' | 'jwt' | 'params'
+  params: Openid4vpAuthRequest
+}
+
+export function parseOpenid4vpRequestParams(
+  input: unknown
+): ParsedOpenid4vpAuthRequest | ParsedJarOpenid4vpAuthRequest {
+  let params = input
+  let provided: 'uri' | 'jwt' | 'params' = 'params'
+
+  if (typeof input === 'string') {
+    if (input.includes('://')) {
+      const data = input.split('://')[1]
+      params = uriDecodeObject(data)
+      provided = 'uri'
+    } else {
+      const decoded = decodeJwt({ jwt: input })
+      params = decoded.payload
+      provided = 'jwt'
+    }
+  }
+
+  if (v.is(vOpenid4vpAuthRequest, params)) {
+    return {
+      type: 'openid4vp',
+      provided,
+      params: v.parse(vOpenid4vpAuthRequest, params),
+    }
+  }
+
+  if (v.is(vJarAuthRequest, params)) {
+    return {
+      type: 'jar',
+      provided,
+      params,
+    }
+  }
+
+  throw new Oauth2Error('Could not parse openid4vp auth request params.')
```

Comment:
I think here somehow including the error messages of what was tried can help in debugging. Especially since if you have one wrong param (let's say you mistyped nonce as nince) it will fail the `v.is()` and thus give this error, but you'd have no idea what went wrong.

Maybe we can do something like a v.union(), add the two types and try to parse it? Then if both fail we have a nice error message of the structures that were tried. The more we do with valibot the less context we lose.